zerver/lib/webhooks/interfaced_settings.py

```diff
@@ -0,0 +1,75 @@
+from dataclasses import asdict, dataclass
+from inspect import signature
+from typing import Annotated, Any, TypeAlias, get_args
+
+from zerver.data_import.sequencer import NEXT_ID
+from zerver.lib.integrations import WebhookIntegration
+from zerver.lib.typed_endpoint import ApiParamConfig
+
+SettingContextT: TypeAlias = dict[str, Any]
+InterfacedSettingT: TypeAlias = dict[str, SettingContextT | None]
+
+
+def setting_prefix_key(param: str) -> str:
+    # This is to make sure interfaced settings query are unique to
+    # other query the integration might declare.
+    param_id = NEXT_ID("param")
+    key = f"{param}_{param_id}"
+    return key
+
+
+MapToChannelsT: TypeAlias = Annotated[bool, ApiParamConfig(setting_prefix_key("map_to_channels"))]
+
+# Keep this in sync with `state_data.integrations_interfaced_settings_schema`
+# this dict is not a dataclass because Type aliases inside dataclass definitions
+# are not supported at runtime.
+SUPPORTED_INTERFACED_SETTINGS: dict[str, TypeAlias] = {"MapToChannelsT": MapToChannelsT}
+
+
+@dataclass
+class SettingContext:
+    parameter_name: str
+    unique_query: str
+
+    def __init__(self, arg_name: str, arg_type: Any) -> None:
+        self.parameter_name = arg_name
+        api_param_config = get_args(arg_type)[1]
+        assert isinstance(api_param_config, ApiParamConfig)
+        whence = api_param_config.whence
+        assert whence is not None
+        self.unique_query = whence
+
+
+def get_settings_context(arg_name: str, arg_type: Any) -> SettingContextT | None:
+    settings_context = None
+    if arg_type == SUPPORTED_INTERFACED_SETTINGS["MapToChannelsT"]:
+        settings_context = SettingContext(arg_name, arg_type)
+
+    if settings_context is not None:
+        return asdict(settings_context)
+    raise AssertionError(f"Please add context for this setting: {arg_name}")
```

Comment:
`Please define a SettingContext for this setting: {arg_name}` would be a lot more clear.